### ISSUE TITLE
[REQUEST] - Will Populate Loot Via Table macro work with Better Roll Tables Module?

### ISSUE THREAD

**Issue by theMagius0** [fantasycalendar/FoundryVTT-ItemPiles#434]
**Is your feature request related to a problem? Please describe.**
I was wondering if it's possible to use the **Populate Loot Via Table** macro in conjunction with the **Better Roll Tables** Module?

**Describe the solution you'd like**
I'm looking to have the macro make a set number of rolls to the "better" table which, in turn, will make its own rolls for multiplicity.

As an example, if I make a macro call to a table with weapons, I might get the result for "arrows".   The "better" table has a multiplicity roll within it such that, if the original roll lands on "arrows", the table rolls 1d6 for the *number* of arrows.  However, when I make the call with a **Populate Loot** macro, I only get a multiplicity of 1--no matter what formula the "better" table has for multiplicity.

**Describe alternatives you've considered**
I have tried making macro calls to "better" tables, but the multiplicity (on the better table) doesn't seem to trigger when I do so.

**Additional context**
None.

**Comment by Haxxer** [comment]
@theMagius0 I have managed to add rudimentary support for the module, but as I do not use it, I don't know all of the use cases. Could you perhaps outline all of the different uses of this module when it comes to generating lists of items?

**Comment by theMagius0** [comment]
Thank you, @Haxxer for the work to make this compatible with BRT.

The author's site lists 4 features:
- Roll on multiple tables with roll formulas
- Auto create a loot actor to store generated loot
- Auto roll random spells when a scroll is selected as loot
- A table can specify multiple currencies (with roll formulas) to always be awarded

But, if I'm being honest, I only use the last one.

My specific use case is that I use your **Populate Loot Via Table** script to make a roll on one (or more) of my tables for, say, ammunition.  But, no matter what I roll, I only get (quantity) 1 of that ammo.  And, so, if I want to populate my loot with multiple ammo quantities of the same round (say, 10mm), I use your script to roll *many* times on the same table.  But, the issue there is that I get "onsie/twosie" ammo quantities, because rolling on a table of 20+ ammo types, I'm not guaranteed to same the *same* ammo more than once.  (see attached pic of what a default Foundry Roll table looks like for ammo).

![BRT_Example01](https://github.com/fantasycalendar/FoundryVTT-ItemPiles/assets/85370867/e19f6f48-547f-4db8-9d82-90a1f010eb7f)

If I create a Better Roll Table, I can specify that, when I roll on the table for a certain ammo type (say, 10mm), instead of a single round, the Better Roll Table will make an additional roll for the *quantity* of that ammo to award.  If I could get this to work with your **Populate Loot Via Table** script, then I could make a single roll, and award more than just a single round.  (see attached pic of what a Better Roll Table looks like for ammo).

![BRT_Example02](https://github.com/fantasycalendar/FoundryVTT-ItemPiles/assets/85370867/6c32fa55-55a1-416d-b8b4-fce55a61ddfd)

So, TLDR, I'd like to be able to randomly populate multiple quantities of the same item using your script.

Thx,

**Comment by Haxxer** [comment]
Added support in 2.7.10

**Comment by dxwarlock** [comment]
With 2.7.11 and the better roll tables, it does not seem to obey/use the total count. I made the table with 1 entry with 2d4 results. No matter how many times I 'Roll all tables" I get one result: 

![image](https://github.com/fantasycalendar/FoundryVTT-ItemPiles/assets/520536/8860b461-9218-4b24-a0ac-5ad38c03d000)
